README
Installation is simple:
For every _* file/directory in here, you want to make it a .* file/directory in your homedir.
Pay attention not to overwrite existing files in the process


ChangeLog
2007-09-20  Yann Hodique  <Yann.Hodique@lifl.fr>

	* dotemacs-cust.el: sanest defaults

2007-09-18  Yann Hodique  <Yann.Hodique@lifl.fr>

	* config/bm-config.el: make package optional

2007-04-11  Yann Hodique  <Yann.Hodique@lifl.fr>

	* icicles: package is now maintained separately

2007-04-09  Yann Hodique  <Yann.Hodique@lifl.fr>

	* lib/magpie.el: initial checkin

	* lib/javascript.el: initial checkin

	* dotemacs-cust.el (custom-set-variables): disable auto update for ecb methods

	* config/python-config.el: initial checkin

	* config/cc-config.el: initial checkin

	* config/bm-config.el: initial checkin

2007-03-07  Yann Hodique  <Yann.Hodique@lifl.fr>

	* dotgnus.el: disable broken signature

	* dotemacs.el: add code for using <Multi_key> as an alias for C-x 8

	* config/buffer-config.el (ibuffer-saved-filter-groups): fix circe group

2007-02-08  Yann Hodique  <Yann.Hodique@lifl.fr>

	* dotemacs.el: added useful wrapper to recursive-edit

	* config/org-config.el (org-agenda): don't minimize ecb compilation window

2007-01-27  Yann Hodique  <Yann.Hodique@lifl.fr>

	* lib/psvn.el: update from mainstream

	* lib/outline-magic.el: initial checkin

	* lib/lua-mode.el: initial checkin

	* lib/blank-mode.el: initial checkin

	* dotemacs.el (savehist): save kill-ring accross sessions

	* custom/autoloads.el (autoload): autoload blanc-mode and predictive

	* config/winring-config.el (eval-when-compile): fix compilation

	* config/outline-config.el: initial checkin

	* config/latex-config.el (outline-config): configure outline for LaTeX-mode

	* config/buffer-config.el (eval-when-compile): fix compilation

2006-12-19  Yann Hodique  <Yann.Hodique@lifl.fr>

	* lib/paredit.el: upgrade to latest beta

	* dotemacs.el: add viewport scrolling
	setup *scratch* buffer from the start

	* custom/patches.el (add-mhook): add &optional arguments

	* config/winring-config.el (winring): fix loading order

	* config/lisp-config.el (yh/lisp-hook): remove reference to sexp hilighting

2006-12-12  Yann Hodique  <Yann.Hodique@lifl.fr>

	* dotemacs-cust.el (custom-set-variables): ignore newlines in regexp searches

	* custom/mycompletion.el (auto-insert-alist): fix C(++) headers

	* config/winring-config.el: configuration for winring

2006-11-25  Yann Hodique  <Yann.Hodique@lifl.fr>

	* custom/patches.el (backup-directory-alist): disable tramp backups

	* lib/incr.el: initial checkin

2006-09-11  Yann Hodique  <Yann.Hodique@lifl.fr>

	* lib/paredit.el: update to latest version

	* dotemacs.el: write windmove bindings in an obfuscated way ;)

	* custom/patches.el: fix strange behavior of mouse-insert-selection-internal

	* config/circe-config.el (yh/lui-truncate): customize truncation

2006-08-03  Yann Hodique  <Yann.Hodique@lifl.fr>

	* lib/psvn.el: new version

	* lib/paredit.el: new version

	* custom/patches.el: looks like RET does not work anymore in custom buffers...

	* config/lisp-config.el (paredit): unbind the <modifier-left/right>
	(slime-repl-mode-hook): use paredit

	* config/circe-config.el (circe-command-RECOVER): add release...
	(circe-clear): truncate buffer to lower value

2006-08-01  Yann Hodique  <Yann.Hodique@lifl.fr>

	* lib/lib-loaddefs.el: updated to reflect cleanup

	* dotgnus.el: using org-mode now

	* dotemacs.el (minimal): load circe

	* config/tabbar-config.el (yh/tabbar-buffer-groups): add case for IRC buffers

	* config/circe-config.el: various hacks

2006-07-28  Yann Hodique  <Yann.Hodique@lifl.fr>

	* config/circe-config.el: extended sorting tracking
	(lui-track-set-modified-status): faces are optional
	ported patches to newer circe

	* config/slime-config.el (slime): install slime48

2006-07-19  Yann Hodique  <Yann.Hodique@lifl.fr>

	* custom/patches.el (temporary-file-directory): bug fix

	* config/slime-config.el: initial checkin

2006-07-06  Yann Hodique  <Yann.Hodique@lifl.fr>

	* lib/icomplete+.el (read-from-minibuffer): fix arguments to comply with new signature

	* config/compile-config.el: don't make compile-history buffer-local after all

	* dotemacs.el: removing config for packages I don't use for now
	more doc

	* dotemacs-cust.el: remove semantic configuration

	* custom/autoloads.el: xtla does not exist anymore

	* config/latex-config.el (request): fix TeX-data-directory for my installation

2006-06-26  Yann Hodique  <Yann.Hodique@lifl.fr>

	* lib/mime.el (mime-mode): remove faulty comment (bug in here)

	* config/ediff-config.el (require): fix compilation issue

	* config/compile-config.el: fix compilation issue

2006-06-25  Yann Hodique  <Yann.Hodique@lifl.fr>

	* perso/patches.el ([insert]): undefine it

	* perso/mycode.el: comment out problematic font-locking

	* latex-style/beamer.el: initial checkin

2006-06-08  Yann Hodique  <Yann.Hodique@lifl.fr>

	* dotemacs.el (kbd): C-c k is now irrelevant
	(emms): new setup for emms 2

	* config/lisp-config.el (yh/insert-elisp-key): function to ease key definition

2006-06-04  Yann Hodique  <Yann.Hodique@lifl.fr>

	* perso/visual.el (default-frame-alist): leave only common configuration
	(window-system-default-frame-alist): specific settings here

	* config/org-config.el (org-insert-heading): advice this function instead

2006-05-30  Yann Hodique  <Yann.Hodique@lifl.fr>

	* perso/physical-line.el: initial checkin

	* perso/paredit.el: update to newest beta

2006-05-29  Yann Hodique  <Yann.Hodique@lifl.fr>

	* dotemacs.el (org-config): load by default
	trigger (org-agenda-list) at startup
	(file-exists-p): absolute filenames in non-local config is *bad* :)

	* config/planner-config.el (kbd): move keybindings to config file

	* config/org-config.el: initial checkin.

	* config/calendar-config.el: disable those useless holidays
	(diary-cyclic-bounded): small variation of diary-cyclic

2006-05-05  Yann Hodique  <Yann.Hodique@lifl.fr>

	* perso/tempo-c++.el (c-mode-common-hook): use common hook

	* perso/paredit.el: initial checkin

	* dotemacs.el: various fixes

	* config/lisp-config.el (paredit-mode): activate paredit-mode


dotemacs.el
;; -*- mode: emacs-lisp; mode: hi-lock; auto-compile-lisp: nil; -*-
;; $Id: dotemacs.el 230 2007-04-09 09:10:18Z yann $

;; (toggle-debug-on-error)

;;;_* Basis

;; load code specific to some major version
(if (file-exists-p (expand-file-name (format "~/.emacs-%d" emacs-major-version)))
    (load-file (expand-file-name (format "~/.emacs-%d" emacs-major-version))))

;; load code specific to some minor version
(if (file-exists-p (expand-file-name (format "~/.emacs-%d-%d" emacs-major-version emacs-minor-version)))
    (load-file (expand-file-name (format "~/.emacs-%d-%d" emacs-major-version emacs-minor-version))))

;; Load site-specific stuff: paths, accounts, passwords, projects...
(if (file-exists-p (expand-file-name "~/.emacs-local"))
    (load-file (expand-file-name "~/.emacs-local")))

;; Fix various "bad" default behaviors
;; add some personal features
(require 'patches)

;; How emacs should look like
(request 'visual)

;; Customizations are in a separate file
(if (file-exists-p (expand-file-name "~/.emacs-cust"))
    (load-file (expand-file-name "~/.emacs-cust")))

;; Hacked scroll margin
(set-scroll-margin 5 5 '("*eshell*" "*compile*" "*Calendar*"))
(setq scroll-step 1
      scroll-conservatively 50)
;; (setq scroll-margin 5
;;       scroll-conservatively 50
;;       scroll-step 1)


;; Save minibuffer history between sessions
(when (request 'savehist)
  (savehist-mode 1)
  (add-to-list 'savehist-additional-variables 'kill-ring))

;; Date in mode line
(display-time)

;; Set window title according to buffer name
(setq frame-title-format '("emacs: %f"))

(when (request 'time-stamp)
  ;; Turn on time stamping
  (setq time-stamp-active t)
  ;; Sets new format for the time stamp, also used with the creation tag.
  (setq time-stamp-format "%02d/%02m/%:y %02H:%02M:%02S %U")
  (add-hook 'before-save-hook 'time-stamp))

;; Default mode
(setq default-major-mode 'indented-text-mode)
(add-hook 'indented-text-mode-hook 'turn-on-auto-fill)

;; We don't want to insert newlines when reaching end of buffer
;; and we want to kill the whole line when doing a C-k
(setq next-line-add-newlines nil
      kill-whole-line t
      kill-read-only-ok t)

(setq-default comment-style 'extra-line)

;; Auto-reload file when modified from external app
(global-auto-revert-mode 1)

;; Enable dynamic highlighting
(if (functionp 'global-hi-lock-mode)
            (global-hi-lock-mode 1)
          (hi-lock-mode 1))

;; Throw away the mouse when typing
(mouse-avoidance-mode 'exile)

;; Load the emacs type verifier (gnu emacs, xemacs, ...)
(request 'emacs-type)

;; Save place by default
(when (request 'saveplace)
  (setq-default save-place t))

;; Use Gnus for mail
(setq mail-user-agent 'gnus-user-agent)
(setq read-mail-command 'gnus)

;; Load default classes
(request 'emacs-d-vars)

;; open compressed files
(request 'jka-compr)
(if (not (jka-compr-installed-p)) (auto-compression-mode))

;; ignore case for finding files
(setq read-file-name-completion-ignore-case t)

(require 'autoloads)

;;;_* Charsets & languages

(add-to-list 'ispell-dictionary-alist '("latin" "[A-Za-z]" "[^A-Za-z]" "[']" nil nil "~tex" iso-8859-1))
(setq ispell-program-name "aspell")

;; (set-language-environment 'latin-1)
;; (prefer-coding-system 'latin-1)
(prefer-coding-system 'mule-utf-8)

(setq unibyte-display-via-language-environment t)
(setq-default ctl-arrow 'latin-9)

(when (request 'ucs-tables)
  (progn
    (unify-8859-on-encoding-mode 1)
    (unify-8859-on-decoding-mode 1)))

(defun sk-insert-euro (&optional arg) "Insert €"
  (interactive "*P")
  (if arg
      (insert (make-char 'mule-unicode-0100-24ff 116 76))
    (insert (make-char 'latin-iso8859-15 164))))

(global-set-key (kbd "H-5") 'sk-insert-euro)

(defun sk-insert-oe (&optional arg) "Insert œ"
  (interactive "*P")
  (insert (make-char 'latin-iso8859-15 #xBD)))

(global-set-key (kbd "H-o H-e") 'sk-insert-oe)

;;;_* Packages configuration

;; I use emacs in 4 different ways from the command line:
;; - as a mail reader (gnus)
;; - as my $EDITOR (aliased as "vi" :p)
;; - as a proof environment (ProofGeneral)
;; - as my default coding environment
(defun current-configuration ()
  (cond ((member "gnus" command-line-args) 'mail)
        ((equal (car command-line-args) "vi") 'minimal)
        ((member "proof-splash-display-screen" command-line-args) 'proof)
        (t 'code)))

(defmacro when-configuration (config &rest body)
  "run BODY when in given configuration"
  (declare (indent 1))
  `(when (equal ,config (current-configuration))
     ,@body))

(defmacro unless-configuration (config &rest body)
  "run BODY unless in given configuration"
  (declare (indent 1))
  `(unless (equal ,config (current-configuration))
     ,@body))

;; common
(request 'buffer-config)
(request 'outline-config)
(request 'org-config)
(request 'calendar-config)

(unless-configuration 'proof
  (request 'ido-config))

;; unless minimal
(unless-configuration 'minimal
  (unless-configuration 'proof
    ;; (request 'muse-config)
    (request 'ecb-config)
    (request 'winring-config)
    ;; (request 'planner-config)
    (request 'circe-config)
    (request 'tramp-config)
    (request 'eshell-config)
    (request 'help-config)
    (request 'moccur-config)
    ;; (request 'lispy-config)
    (request 'tabbar-config)
    ;; (request 'sawfish-config)
    ;; (request 'pmwiki-config)
    (request 'psvn-config)))

(when-configuration 'code
  (request 'bm-config)
  (request 'changelog-config)
  (request 'cc-config)
  (request 'compile-config)
  (request 'ediff-config)
  ;; (request 'speedbar-config)
  ;; (request 'doxymacs-config)
  (request 'latex-config)
  ;; (request 'highlight-changes-config)
  ;; (request 'fracc-config)
  (request 'paren-config)
  (request 'lisp-config)
  (request 'slime-config)
  (request 'shell-config)
  (request 'python-config)
  (request 'mycode)
  (request 'project-config)
  (request 'completion-config)
  (request 'crontab-config)
  ;; (request 'flashcard-config)
  (request 'vc-config))

;; _* Utils/Functions

(make-double-command my-home ()
  "Go to beginning of line, or beginning of buffer."
  nil
  (beginning-of-line)
  (beginning-of-buffer))

(make-double-command my-end ()
  "Go to end of line, or end of buffer."
  nil
  (end-of-line)
  (end-of-buffer))

;; byte-compile current elisp buffer
(defun byte-compile-elisp ()
  "Byte compile the current buffer if possible"
  (if (and auto-compile-lisp
           (not (backup-file-name-p (buffer-file-name))))
      (cond
       ((eq major-mode 'sawfish-mode)
        (sawfish-compile-this-file))
       ((eq major-mode 'emacs-lisp-mode)
        (byte-compile-file (buffer-file-name)))
       )))

;; make sure modifs are taken into account (use with caution)
(add-hook 'after-save-hook 'byte-compile-elisp)

;; make my scripts executable
(add-hook 'after-save-hook 'executable-make-buffer-file-executable-if-script-p)

;; I hate trailing whitespaces (use with caution)
(add-hook 'write-file-hooks 'delete-trailing-whitespace)

;; I also hate additional blank-lines (use with extreme caution)
;;(add-hook 'write-file-hooks 'simplify-blank-lines)

;; Adaptative "exit" behavior
(defun exit-no-gnuserv-or-frame ()
  "If in a gnuserv-edit session, close it. If in a secondary frame, close it. Else, die"
  (interactive)
  (cond ((and (boundp 'gnuserv-minor-mode) gnuserv-minor-mode) (gnuserv-edit))
	((or (not main-frame) (eq (selected-frame) main-frame)) (save-buffers-kill-emacs))
	((delete-frame))))

;; make active frame the main one
(defun make-main-frame ()
  "Make the current frame the primary one"
  (interactive)
  (setq main-frame (selected-frame)))

;; My init function for main window
(defun init ()
  (interactive)
  (when (request 'ecb)
    (let ((act (request 'winring)))
      (ecb-activate)
      (when act
	(progn
	  (ecb-winman-winring-enable-support)
	  (winring-initialize))))))

(request 'server)
(defun root-portal ()
  "Function to populate background"
  (interactive)
  ;; frame settings
  (xterm-mouse-mode 1)
  ;; prepare
  (delete-other-windows)
  ;; need to start this first for calendar buffer to end up lowest
  (calendar)
  ;; build window configuration
  (other-window 1)
  (split-window-horizontally)
  (other-window 1)
  ;; terminal window width
  (enlarge-window-horizontally
   (- 80 (window-width)))
  (split-window-vertically)
  (ansi-term "/bin/bash" "top")
  (term-exec (current-buffer) "top" "top" nil nil)
  (other-window -1)
  (find-file (concat planner-directory "/" planner-default-page))
  (split-window-vertically)
  (set-window-text-height (selected-window) 72)
  (other-window 1)
  (eshell)
  (other-window -1))

(make-main-frame)
(init)

;; tidy up diffs when closing the file
(defun kill-associated-diff-buf ()
  (let ((buf (get-buffer (concat "*Assoc file diff: "
                             (buffer-name)
                             "*"))))
    (when (bufferp buf)
      (kill-buffer buf))))

(add-hook 'kill-buffer-hook 'kill-associated-diff-buf)

;;;_* Global key bindings

(global-set-key (kbd "<C-backspace>") 'kill-syntax-backward)

(global-set-key (kbd "C-c +") 'incr-dwim)

(global-set-key (kbd "<f3>") 'ecb-toggle-compile-window)
;; Depending on your keyboard you may want another one binding
(global-set-key (kbd "C-x ~") 'previous-error)
(global-set-key (kbd "C-c s") 'eshell)
(global-set-key (kbd "C-c c") 'compile)
(global-set-key (kbd "C-c g") 'goto-line)
(global-set-key (kbd "C-c o") 'my-occur)
(global-set-key (kbd "C-c u") 'remember)
(global-set-key (kbd "C-c e") 'fc-eval-and-replace)
(global-set-key (kbd "C-c f") 'find-function)
(global-set-key (kbd "C-c F") 'find-function-on-key)

;; Enter a recursive edit. C-M-c will bring back exactly there
(global-set-key (kbd "C-c r") (lambda ()
                                (interactive)
                                (save-window-excursion
                                  (save-excursion
                                    (recursive-edit)))))

(global-set-key (kbd "<C-tab>") 'other-window)

;; These were traditional bindings, why did they change??
(global-set-key (kbd "<home>") 'beginning-of-buffer)
(global-set-key (kbd "<end>") 'end-of-buffer)
(global-set-key (kbd "<delete>") 'delete-char)

;; "intelligent" home and end
(global-set-key (kbd "<C-home>") 'my-home)
(global-set-key (kbd "<C-end>") 'my-end)

;; mouse scrolling
(unless (featurep 'mwheel)
  (global-set-key (kbd "<mouse-4>") 'scroll-down)
  (global-set-key (kbd "<mouse-5>") 'scroll-up)
  (global-set-key (kbd "<C-mouse-4>") 'scroll-down-one-line)
  (global-set-key (kbd "<C-mouse-5>") 'scroll-up-one-line))

;; viewport scrolling

(global-set-key (kbd "<M-up>")
                (lambda (arg)
                  (interactive "p")
                  (or arg (setq arg 1))
                  (let ((top (line-number-at-pos (window-start)))
                        (cur (line-number-at-pos (point))))
                    (when (>= (- cur top) arg)
                      (scroll-up arg)))))

(global-set-key (kbd "<M-down>")
                (lambda (arg)
                  (interactive "p")
                  (or arg (setq arg 1))
                  (let ((cur (line-number-at-pos (point)))
                        (bot (+ (line-number-at-pos (window-start))
                                (window-text-height)
                                -1)))
                    (when (>= (- bot cur) arg)
                      (scroll-down arg)))))

;; rectangle bindings. don't mix with registers! :)
(global-set-key (kbd "C-x r C-SPC") 'rm-set-mark)
(global-set-key (kbd "C-x r C-x") 'rm-exchange-point-and-mark)
(global-set-key (kbd "C-x r C-w") 'rm-kill-region)
(global-set-key (kbd "C-x r M-w") 'rm-kill-ring-save)
(global-set-key (kbd "C-x r C-y") 'yank-rectangle)

(global-set-key (kbd "C-x C-c") 'exit-no-gnuserv-or-frame)
(global-set-key (kbd "C-x 5 3") 'make-main-frame)
(global-set-key (kbd "C-c d") 'diff-buffer-with-associated-file)
(global-set-key (kbd "C-x k") 'de-context-kill)
(global-set-key (kbd "C-x K") (lambda () (interactive) (dolist (buf (buffer-list)) (when (buffer-file-name buf) (kill-buffer buf)))))

(global-set-key (kbd "C-c x") 'chmod-file)
(global-set-key (kbd "C-c i") 'init)
(global-set-key (kbd "C-c h") 'auto-insert)
;;(global-set-key (kbd "C-x C-f") 'find-file-guessing)

(global-set-key (kbd "H-s") 'isearchb-activate)

(define-key isearch-mode-map (kbd "C-o")
  (lambda ()
    (interactive)
    (let ((case-fold-search isearch-case-fold-search))
      (occur (if isearch-regexp isearch-string
               (regexp-quote isearch-string))))))

(global-set-key (kbd "<f5>")
                (lambda () (interactive)
                  (if (and (boundp 'ecb-minor-mode) ecb-minor-mode)
                      (ecb-deactivate)
                    (ecb-activate))))

(global-set-key (kbd "H-z") 'zap-upto-char)
(global-set-key (kbd "H-M-z") 'zap-to-char)
(global-set-key (kbd "H-Z") 'zap-following-char)
(global-set-key (kbd "H-M-Z") 'zap-from-char)

;; Project related keys
(global-set-key (kbd "<C-f1>") 'toggle-source-header)

;; replace a recently typed wrong character
(global-set-key (kbd "H-r")  'replace-recent-character)

;; use multiple selections
(when (request 'multi-region)
  (global-set-key (kbd "<H-return>") multi-region-map))

;; versioning keys
(global-set-key (kbd "<M-f12>") 'svn-status)
(global-set-key (kbd "<C-f12>") 'cvs-update)

;;; windmove :)
(mapc #'eval
      (mapcar #'(lambda (dir)
                  (let ((name (symbol-name dir)))
                    `(global-set-key (kbd ,(format "C-x <%s>" name))
                                     ',(intern (concat "windmove-" name)))))
              '(left right up down)))

;; xterm settings
(when (and (string= "xterm" (getenv "TERM"))
           (request 'xterm-extras))
  (xterm-extra-keys))

(request 'w3m-load)

;;;_* Experimental

;; Doc-mode
(require 'doc-mode)
(add-hook 'c++-mode-hook 'doc-mode)
(add-hook 'java-mode-hook 'doc-mode)

;; Compagny-mode
(require 'company-mode)
(require 'company-bundled-completions)
(company-install-bundled-completions-rules)

;; Just in case compose is broken...
(define-key key-translation-map (kbd "<Multi_key>") 'iso-transl-ctl-x-8-map)
(autoload 'iso-transl-ctl-x-8-map "iso-transl" "Keymap for C-x 8 prefix." t 'keymap)

(when (request 'pp-c-l)
  (pretty-control-l-mode 1))

(autoload 'graphviz-dot-mode "graphviz-dot-mode" "" t nil)
(add-to-list 'auto-mode-alist '("\\.dot\\'" . graphviz-dot-mode))

(add-to-list 'auto-mode-alist '("\\.hlal\\'" . c-mode))

(autoload 'javascript-mode "javascript" nil t)
(add-to-list 'auto-mode-alist '("\\.js\\'" . javascript-mode))

(eval-after-load "javascript"
  '(add-hook 'javascript-mode (lambda ()
                                (glasses-mode 1)
                                (c-subword-mode 1))))

(when (request 'incr)
  (delq 'rotate incr-enable-feature))

;; (setq sgml-warn-about-undefined-entities nil)

;;************************************************************
;; configure HTML editing
;;************************************************************

(autoload 'php-mode "php-mode" "" t nil)
(add-to-list 'auto-mode-alist '("\\.php[34]?\\'" . php-mode))

(add-to-list 'auto-mode-alist '("\\.css\\'" . css-mode))
(setq cssm-indent-function #'cssm-c-style-indenter)
(setq cssm-indent-level '2)

(add-hook 'php-mode-user-hook 'turn-on-font-lock)

(add-to-list 'auto-mode-alist '("\\.[hi]\\'" . c++-mode))
(add-to-list 'auto-mode-alist '("\\.djava\\'" . java-mode))
(add-to-list 'auto-mode-alist '("\\.pro\\'" . makefile-mode))

(global-set-key (kbd "C-x t") 'anchored-transpose)
(autoload 'anchored-transpose "anchored-transpose" nil t)

(when-configuration 'code
  (try (server-mode 1)))

;;; Startup code
(require 'org)
(when (and (boundp 'org-default-notes-file)
	   (file-exists-p org-default-notes-file))
  (find-file org-default-notes-file)
  (require 'calendar)
  (call-interactively 'org-agenda-list)
  (eshell))

(global-hl-line-hack-mode 1)

(autoload 'predictive-mode "predictive" "predictive" t)
(set-default 'predictive-auto-add-to-dict t)
(setq predictive-main-dict 'dict-english
      predictive-auto-learn t
      predictive-add-to-dict-ask nil
      predictive-use-auto-learn-cache nil
      predictive-which-dict t)

;; setup *scratch* correctly
(kill-scratch-buffer)

(message ".emacs loaded")
